docs/reference/vpc-association-policy.md

```diff
@@ -0,0 +1,85 @@
+# VpcAssociationPolicy API Reference
+
+## VpcAssociationPolicy
+
+VpcAssociationPolicy is a CRD that can be attached to a Gateway to define the ServiceNetworkVpcAssociation configuration.
+
+
+
+One of its option is `securityGroupIds`. it can control the inbound traffic from current cluster workloads to the gateway listeners. Please check the VPC lattice doc for more detail of this option. https://docs.aws.amazon.com/vpc-lattice/latest/ug/security-groups.html
+
+Recommended security group inbound rules:
+
+| Source                                                  | Protocol                                            | Port Range                                      | Comment                                                   |
+|---------------------------------------------------------|-----------------------------------------------------|-------------------------------------------------|-----------------------------------------------------------|
+| Kubernetes cluster VPC CIDR or security group reference | protocols defined in the gateway's listener section | ports defined in the gateway's listener section | Allow inbound traffic from current cluster vpc to gateway |
+
+
+**Limitations and Considerations**
+
+When attaching a policy to a resource, the following restrictions apply:
+
+* A policy can be only attached to `Gateway` resources.
+* The attached resource should exist in the same namespace as the policy resource.
+
+The security Group will not take effect if:
+
+* The targetRef `Gateway` does not exist
+* AssociateWithVpc field set to false
+
+
+**WARNING**
+
+Current VPC Lattice updateServiceNetworkVpcAssociation api have a limitation that it cannot remove all security groups.
+That means, if you have a VpcAssociationPolicy attached to a gateway that already applied security groups, following operations will NOT take effect to remove the security groups:
+* Update the VPCAssociationPolicy to empty the security group ids (even though the updated VPCAssociationPolicy can be accepted by the API server)
+* Delete the VPCAssociationPolicy (even though the VPCAssociationPolicy can be deleted from k8s successfully)
```

Comment:
Just to clarify, if the customer tries these steps, it will NOT work? If so, we should not list them as part of the documentation at all, and instead just give guidance on what they should do.